src/constructs/lambda/lambda.ts

```diff
@@ -21,12 +22,28 @@ interface GuFunctionProps extends Omit<FunctionProps, "code"> {
   apis?: ApiProps[];
 }
 
+function defaultMemorySize(runtime: Runtime, memorySize?: number): number {
+  if (memorySize) {
+    return memorySize;
+  } else {
+    switch (runtime.family) {
+      case RuntimeFamily.JAVA:
+        return 1024;
+      default:
+        return 512;
+    }
+  }
+}
+
 export class GuLambdaFunction extends Function {
   constructor(scope: GuStack, id: string, props: GuFunctionProps) {
     const bucket = Bucket.fromBucketName(scope, `${id}-bucket`, props.code.bucket);
     const code = Code.fromBucket(bucket, props.code.key);
+    const { memorySize, timeout, ...rest } = props;
```

Comment:
Do these still need to be extracted from props or would it be easier just to use `props.memorySize` on line 45?

Reply:
Ah yes thanks, I didn't realise this simpler version would work (but now I've read the docs and re-tested I can see that my version was needlessly complicated 😀).